File: Kms.Security.Portal/app/nodak/common/dynamicPermission/dynamicPermission.ts
module nodak.common.models {
    export class DynamicPermission extends nodak.models.ModelBase {
        constructor() {
            super(enums.SubSystems.Common, "DynamicPermission");
            //this.model = new kms.userManagement.models.MemberDto();
            //this.HasProperty(() => this.LastName, "نام خانوادگی").IsRequired();
            //this.HasProperty(() => this.FirstName, "نام ").IsRequired();
            //this.HasProperty(() => this.Password, "رمز عبور ").IsRequired();
            //this.HasProperty(() => this.UserName, "نام کاربری ").IsRequired();
            //this.HasProperty(() => this.Email, "ایمیل ").IsRequired();
        }
        AddBussinessModelValidation() { }
        private id: string;
        set Id(value) {
            this.id = value;
        }
        get Id() {
            return this.id;
        }
        //=============
        private name: string;
        set Name(value) {
            this.name = value;
        }
        get Name() {
            return this.name;
        }

        private title: string;
        get Title() {
            return this.title;
        }
        set Title(value) {
            this.title = value;
        }

        private parentId: string;
        get ParentId() {
            return this.parentId;
        }
        set ParentId(value) {
            this.parentId = value;
        }

        private controllerName: string;
        set ControllerName(value) {
            this.controllerName = value;
        }
        get CoontrollerName() {
            return this.controllerName;
        }


    }
}

File: Kms.Security.Common/Domain/DynamicPermission.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kms.Security.Common.Domain

{
    public class DynamicPermission
    {
        public string ControllerName { get; set; }
        public string Id
        {
            get
            {
                return ControllerName + "_" + Name;
            }
        }
        public string Name { get; set; }
        public string ReturnType { get; set; }
        public string Title { get; set; }

        public string ParentId { get; set; }
    }
}
